### ISSUE TITLE
API broken for all augmentations for the probability "p" argument

### ISSUE THREAD

**Issue by MrRss** [kornia/kornia#859]
## 🐛 Bug

The probability argument (p) of the kornia augmentation is never used. The problem seems to consists for all augmentations. 
## To Reproduce

For instance: 
1. Apply RandomHorizontalFlip with p = 0

## Expected behavior
Transform should not be applied, i.e. image should not be flipped.

**Comment by shijianjian** [comment]
@MrRss 

I cannot reproduce this problem on both v0.4.1 and master branch. Can you try to install the master instead? Or please provide some more details.

**Comment by MrRss** [comment]
@shijianjian 

Thanks for your answer. I installed master and have the same behaviour, i.e. when I set p to zero for flip, perspective transform and rotation, they are always applied.
code:

import kornia as K
from kornia.augmentation import AugmentationBase2D
from kornia.constants import Resample
from torch import nn
import torch
from typing import Dict
from kornia import functional as F


class MyGPUTransformPipeline(nn.Module):
    def __init__(self) -> None:
        super(MyGPUTransformPipeline, self).__init__()

        self.rot = K.augmentation.RandomRotation(
            p=1, degrees=180, resample=Resample.NEAREST
        )

        self.persp = K.augmentation.RandomPerspective(
            p=1, distortion_scale=0.3, resample=Resample.NEAREST
        )

        self.scale = K.augmentation.RandomAffine(
            degrees=0, scale=(0.5, 1), resample=Resample.NEAREST, p=1
        )

        self.erase = K.augmentation.RandomErasing(
            scale=(0.0, 0.005),
            ratio=(0.1, 1),
            p=1,
        )

        self.add = K.augmentation.RandomErasing(
            scale=(0.0, 0.005),
            ratio=(0.01, 0.9),
            p=1 / 3,
        )

        self.blur = K.augmentation.RandomMotionBlur(
            p=0.5, kernel_size=3, angle=(0, 180), direction=(-1, 1)
        )

    @torch.no_grad()
    def forward(self, input):

        rot_params = self.rot.generate_parameters(input.shape)
        input = self.rot(input, rot_params)

        persp_params = self.persp.generate_parameters(input.shape)
        input = self.persp(input, persp_params)

        # erase_params = self.erase.generate_parameters(input.shape)
        # input = self.erase(input, erase_params)

        # input = 1 - input
        # add_params = self.add.generate_parameters(input.shape)
        # input = self.add(input, add_params)
        # input = 1 - input

        scale_param = self.scale.generate_parameters(input.shape)
        input = self.scale(input, scale_param)

        blur_params = self.blur.generate_parameters(input.shape)
        input = self.blur(input, blur_params)
        input[input > 0] = 1
        input[input < 0] = 0

        return input


class MyGPUTransformPipelineFlip(nn.Module):
    def __init__(self) -> None:
        super(MyGPUTransformPipelineFlip, self).__init__()
        self.flip_hor = RandomHorizontalFlipFixed(p=0.5)

    @torch.no_grad()
    def forward(self, input):

        flip_hor_params = self.flip_hor.generate_parameters(input.shape)
        input = self.flip_hor(input, flip_hor_params)

        return input

environment

PyTorch version: 1.6.0
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 10 Pro
GCC version: (Rev4, Built by MSYS2 project) 10.2.0
Clang version: Could not collect
CMake version: Could not collect

Python version: 3.7 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: 10.1.243
GPU models and configuration: GPU 0: TITAN RTX
Nvidia driver version: 456.71
cuDNN version: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1\bin\cudnn64_7.dll
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] efficientnet-pytorch==0.6.3
[pip3] numpy==1.19.3
[pip3] segmentation-models-pytorch==0.1.3
[pip3] torch==1.6.0
[pip3] torchvision==0.7.0
[conda] blas                      1.0                         mkl
[conda] cudatoolkit               10.1.243             h74a9793_0
[conda] efficientnet-pytorch      0.6.3                    pypi_0    pypi
[conda] libblas                   3.8.0                    20_mkl    conda-forge
[conda] libcblas                  3.8.0                    20_mkl    conda-forge
[conda] liblapack                 3.8.0                    20_mkl    conda-forge
[conda] liblapacke                3.8.0                    20_mkl    conda-forge
[conda] mkl                       2020.2                      256
[conda] mkl-service               2.3.0            py37h196d8e1_0
[conda] mkl_fft                   1.1.0            py37h45dec08_0
[conda] mkl_random                1.1.1            py37h47e9c7a_0
[conda] numpy                     1.19.3                   pypi_0    pypi
[conda] numpy-base                1.19.1           py37ha3acd2a_0
[conda] segmentation-models-pytorch 0.1.3                    pypi_0    pypi
[conda] torch                     1.6.0                    pypi_0    pypi
[conda] torchvision               0.7.0                    pypi_0    pypi

**Comment by shijianjian** [comment]
I see the problem. The function ```generate_parameters``` is not looking at parameter ```p``` now.

The reason of it is because ```generate_parameters``` can be a user-defined function that we do not want to expose the detailed probability control to them.

If you want to access the parameter, you may use:
```
@torch.no_grad()
def forward(self, input):

    input = self.flip_hor(input)
    flip_hor_params = self.flip_hor._params

    return input
```

or
```
@torch.no_grad()
def forward(self, input):

    flip_hor_params = self.flip_hor.__forward_parameter__(input, p, p_batch=1., same_on_batch=False)
    input = self.flip_hor(input, flip_hor_params)

    return input
```

We will be updating the docs and consider if to have a ```forward_parameter``` for this purpose.

**Comment by MrRss** [comment]
Thanks for your fast answer. Let me check, if I understand it correctly:

In my code above, it would be sufficient to call for every augmentation for instance
`input = self.flip_hor(input)`

instead of 
```
input = self.flip_hor(input)
flip_hor_params = self.flip_hor._params
```
or
```
flip_hor_params = self.flip_hor.__forward_parameter__(input, p, p_batch=1., same_on_batch=False)
input = self.flip_hor(input, flip_hor_params)
```

If I would do segmentation, it should look like this:
```
flip_hor_params = self.flip_hor.__forward_parameter__(input, p, p_batch=1., same_on_batch=False)
mask = self.flip_hor(mask, flip_hor_params)
image= self.flip_hor(image, flip_hor_params)
```

**Comment by shijianjian** [comment]
It is recommended to do as:
```
@torch.no_grad()
def forward(self, input, mask):

    input = self.flip_hor(input)
    mask = self.flip_hor(mask, self.flip_hor._params)

    return input, mask
```

**Comment by MrRss** [comment]
Ok, then also this doc has to be updated I guess: https://kornia.readthedocs.io/en/latest/augmentation.html

**Comment by edgarriba** [comment]
@shijianjian @MrRss maybe it's worth having a getter for params which can be more user-friendly

**Comment by shijianjian** [comment]
Fixed by #885
